pkg/kube/krt/index.go

```diff
@@ -37,8 +37,9 @@ func (i *Index[I, K]) Lookup(k K) []I {
 	for obj := range i.objects[k] {
 		item := i.c.GetKey(obj)
 		if item == nil {
-			log.Errorf("Impossible missing item for %v", obj)
-			// This should be extremely rare, maybe impossible due to the mutex.
+			// This should be extremely rare, but possible. While we have a mutex here, the underlying collection
+			// is not locked and maybe have changed in the meantime.
+			log.Debugf("missing item for %v", obj)
```

Comment:
Is there any point in `continue`-ing if this happens?

Reply:
Yeah we are just missing one, not all of them. This just means one of them was removed but out `i.objects` wasn't yet updated. If we had called 1ms later, it would not be in `i.objects`, but there may be other valid things